
<!-- badges: start -->
[![CRAN](https://www.r-pkg.org/badges/version/clustermole)](https://cran.r-project.org/package=clustermole)
[![R-CMD-check](https://github.com/igordot/clustermole/actions/workflows/R-CMD-check.yaml/badge.svg)](https://github.com/igordot/clustermole/actions/workflows/R-CMD-check.yaml)
[![codecov](https://codecov.io/gh/igordot/clustermole/graph/badge.svg?token=YoTQTU1EDk)](https://codecov.io/gh/igordot/clustermole)
<!-- badges: end -->

Assignment of cell type labels to single-cell RNA sequencing (scRNA-seq) clusters is often a time-consuming process that involves manual inspection of the cluster marker genes complemented with a detailed literature search.
This can be especially challenging when unexpected or poorly described populations are present.
The clustermole R package provides methods to query cell identity markers sourced from a variety of databases.

---

![](https://user-images.githubusercontent.com/6363505/72761156-12414280-3ba9-11ea-87de-57ff6cd690bb.png)
*Image credit: "A Child's Primer Of Natural History" by Oliver Herford*
