<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Introduction to clustermole • clustermole</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/readable/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Introduction to clustermole">
<meta property="og:description" content="clustermole">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-130263433-2"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-130263433-2');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">clustermole</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.0.1.9001</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"></ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="../articles/clustermole-intro.html">Get started</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/articles/stats.html">Database statistics</a>
    </li>
    <li>
      <a href="../articles/clustermole-intro.html">Introduction to clustermole</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
<li>
  <a href="https://github.com/igordot/clustermole/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="clustermole-intro_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Introduction to clustermole</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/igordot/clustermole/blob/master/vignettes/clustermole-intro.Rmd"><code>vignettes/clustermole-intro.Rmd</code></a></small>
      <div class="hidden name"><code>clustermole-intro.Rmd</code></div>

    </div>

    
    
<div id="overview" class="section level2">
<h2 class="hasAnchor">
<a href="#overview" class="anchor"></a>Overview</h2>
<p>A typical computational pipeline to process single-cell RNA sequencing (scRNA-seq) data includes clustering of cells as one of the steps. Assignment of cell type labels to those clusters is often a time-consuming process that involves manual inspection of the cluster marker genes complemented with a detailed literature search. This is especially challenging when unexpected or poorly described populations are present. The clustermole R package provides methods to query thousands of human and mouse cell identity markers sourced from a variety of databases.</p>
<p>The clustermole package provides three primary features:</p>
<ul>
<li>a database of markers for hundreds of cell types (<code>clustermole_markers</code>)</li>
<li>cell type prediction based on marker genes (<code>clustermole_overlaps</code>)</li>
<li>cell type prediction based on the full expression matrix (<code>clustermole_enrichment</code>)</li>
</ul>
</div>
<div id="usage" class="section level2">
<h2 class="hasAnchor">
<a href="#usage" class="anchor"></a>Usage</h2>
<p>You can install clustermole from <a href="https://cran.r-project.org/package=clustermole">CRAN</a>.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"clustermole"</span><span class="op">)</span></code></pre></div>
<p>Load clustermole.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://igordot.github.io/clustermole">clustermole</a></span><span class="op">)</span></code></pre></div>
<div id="clustermole_overlaps-cell-type-prediction-based-on-marker-genes" class="section level3">
<h3 class="hasAnchor">
<a href="#clustermole_overlaps-cell-type-prediction-based-on-marker-genes" class="anchor"></a><code>clustermole_overlaps()</code>: cell type prediction based on marker genes</h3>
<p>If you have a set of genes, such as cluster markers, you can compare them to known cell type markers to see if they overlap any of the known cell type markers (overrepresentation analysis).</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">my_genes</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"CD2"</span>, <span class="st">"CD3D"</span>, <span class="st">"CD3E"</span>, <span class="st">"IL7R"</span>, <span class="st">"IL32"</span>, <span class="st">"LTB"</span>, <span class="st">"LDHB"</span>, <span class="st">"CCR7"</span><span class="op">)</span>
<span class="va">my_overlaps</span> <span class="op">=</span> <span class="fu"><a href="../reference/clustermole_overlaps.html">clustermole_overlaps</a></span><span class="op">(</span>genes <span class="op">=</span> <span class="va">my_genes</span>, species <span class="op">=</span> <span class="st">"hs"</span><span class="op">)</span>
<span class="va">my_overlaps</span>
<span class="co">#&gt; # A tibble: 364 x 9</span>
<span class="co">#&gt;    celltype_full db     species organ celltype  n_genes overlap  p_value     fdr</span>
<span class="co">#&gt;    &lt;chr&gt;         &lt;chr&gt;  &lt;chr&gt;   &lt;chr&gt; &lt;chr&gt;       &lt;int&gt;   &lt;dbl&gt;    &lt;dbl&gt;   &lt;dbl&gt;</span>
<span class="co">#&gt;  1 FAN_EMBRYONI… MSigDB Human   ""    FAN_EMBR…     113       5 3.07e-11 9.33e-8</span>
<span class="co">#&gt;  2 T memory cel… Pangl… Human   "Imm… T memory…      38       4 1.66e-10 1.00e-7</span>
<span class="co">#&gt;  3 Naive T cell… CellM… Human   "Blo… Naive T …       5       3 1.71e-10 1.00e-7</span>
<span class="co">#&gt;  4 T memory cel… Pangl… Mouse   "Imm… T memory…      40       4 2.77e-10 1.00e-7</span>
<span class="co">#&gt;  5 IMGN_T_4Nve_… SaVanT Mouse   ""    IMGN_T_4…      50       4 5.17e-10 1.00e-7</span>
<span class="co">#&gt;  6 IMGN_T_4SP24… SaVanT Mouse   ""    IMGN_T_4…      50       4 5.17e-10 1.00e-7</span>
<span class="co">#&gt;  7 IMGN_T_8Nve_… SaVanT Mouse   ""    IMGN_T_8…      50       4 5.17e-10 1.00e-7</span>
<span class="co">#&gt;  8 IMGN_T_8Nve_… SaVanT Mouse   ""    IMGN_T_8…      50       4 5.17e-10 1.00e-7</span>
<span class="co">#&gt;  9 MA_iTreg_IL-… SaVanT Human   ""    MA_iTreg…      50       4 5.17e-10 1.00e-7</span>
<span class="co">#&gt; 10 MA_Tresting … SaVanT Human   ""    MA_Trest…      50       4 5.17e-10 1.00e-7</span>
<span class="co">#&gt; # … with 354 more rows</span></code></pre></div>
</div>
<div id="clustermole_enrichment-cell-type-enrichment-in-the-full-expression-matrix" class="section level3">
<h3 class="hasAnchor">
<a href="#clustermole_enrichment-cell-type-enrichment-in-the-full-expression-matrix" class="anchor"></a><code>clustermole_enrichment()</code>: cell type enrichment in the full expression matrix</h3>
<p>If you have a table of expression values, such as average expression across clusters, you can perform cell type enrichment based on a given gene expression matrix (log-transformed CPM/TPM/FPKM values). Genes are rows and clusters/samples are columns.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/clustermole_enrichment.html">clustermole_enrichment</a></span><span class="op">(</span>expr_mat <span class="op">=</span> <span class="va">my_expr_mat</span>, species <span class="op">=</span> <span class="st">"hs"</span><span class="op">)</span></code></pre></div>
</div>
<div id="clustermole_markers-retrieve-cell-type-markers" class="section level3">
<h3 class="hasAnchor">
<a href="#clustermole_markers-retrieve-cell-type-markers" class="anchor"></a><code>clustermole_markers()</code>: retrieve cell type markers</h3>
<p>You can use <code>clustermole</code> as a simple database and get a data frame of all cell type markers.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">markers</span> <span class="op">=</span> <span class="fu"><a href="../reference/clustermole_markers.html">clustermole_markers</a></span><span class="op">(</span>species <span class="op">=</span> <span class="st">"hs"</span><span class="op">)</span>
<span class="va">markers</span>
<span class="co">#&gt; # A tibble: 422,292 x 8</span>
<span class="co">#&gt;    celltype_full      db     species organ  celltype   n_genes gene_origi… gene </span>
<span class="co">#&gt;    &lt;chr&gt;              &lt;chr&gt;  &lt;chr&gt;   &lt;chr&gt;  &lt;chr&gt;        &lt;int&gt; &lt;chr&gt;       &lt;chr&gt;</span>
<span class="co">#&gt;  1 1-cell stage cell… CellM… Human   Embryo 1-cell st…      32 ACCSL       ACCSL</span>
<span class="co">#&gt;  2 1-cell stage cell… CellM… Human   Embryo 1-cell st…      32 ACVR1B      ACVR…</span>
<span class="co">#&gt;  3 1-cell stage cell… CellM… Human   Embryo 1-cell st…      32 ASF1B       ASF1B</span>
<span class="co">#&gt;  4 1-cell stage cell… CellM… Human   Embryo 1-cell st…      32 BCL2L10     BCL2…</span>
<span class="co">#&gt;  5 1-cell stage cell… CellM… Human   Embryo 1-cell st…      32 BLCAP       BLCAP</span>
<span class="co">#&gt;  6 1-cell stage cell… CellM… Human   Embryo 1-cell st…      32 CASC3       CASC3</span>
<span class="co">#&gt;  7 1-cell stage cell… CellM… Human   Embryo 1-cell st…      32 CLEC10A     CLEC…</span>
<span class="co">#&gt;  8 1-cell stage cell… CellM… Human   Embryo 1-cell st…      32 CNOT11      CNOT…</span>
<span class="co">#&gt;  9 1-cell stage cell… CellM… Human   Embryo 1-cell st…      32 DCLK2       DCLK2</span>
<span class="co">#&gt; 10 1-cell stage cell… CellM… Human   Embryo 1-cell st…      32 DHCR7       DHCR7</span>
<span class="co">#&gt; # … with 422,282 more rows</span></code></pre></div>
<p>Each row contains a gene and a cell type associated with it. The <code>gene</code> column is the gene symbol (human or mouse versions can be retrieved) and the <code>celltype_full</code> column contains the full cell type string, including the species and the original database.</p>
<p>If you need to convert the markers from a data frame to a list format for other applications, you can use <code>gene</code> as the values and <code>celltype_full</code> as the grouping variable.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">markers_list</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/split.html">split</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">markers</span><span class="op">$</span><span class="va">gene</span>, f <span class="op">=</span> <span class="va">markers</span><span class="op">$</span><span class="va">celltype_full</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div id="database-details" class="section level2">
<h2 class="hasAnchor">
<a href="#database-details" class="anchor"></a>Database details</h2>
<p>We will load dplyr to help with the summary statistics.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org">dplyr</a></span><span class="op">)</span></code></pre></div>
<p>You can use <code><a href="../reference/clustermole_markers.html">clustermole_markers()</a></code> to retrieve a data frame of all cell type markers in the collection.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">markers</span> <span class="op">=</span> <span class="fu"><a href="../reference/clustermole_markers.html">clustermole_markers</a></span><span class="op">(</span>species <span class="op">=</span> <span class="st">"hs"</span><span class="op">)</span>
<span class="va">markers</span>
<span class="co">#&gt; # A tibble: 422,292 x 8</span>
<span class="co">#&gt;    celltype_full      db     species organ  celltype   n_genes gene_origi… gene </span>
<span class="co">#&gt;    &lt;chr&gt;              &lt;chr&gt;  &lt;chr&gt;   &lt;chr&gt;  &lt;chr&gt;        &lt;int&gt; &lt;chr&gt;       &lt;chr&gt;</span>
<span class="co">#&gt;  1 1-cell stage cell… CellM… Human   Embryo 1-cell st…      32 ACCSL       ACCSL</span>
<span class="co">#&gt;  2 1-cell stage cell… CellM… Human   Embryo 1-cell st…      32 ACVR1B      ACVR…</span>
<span class="co">#&gt;  3 1-cell stage cell… CellM… Human   Embryo 1-cell st…      32 ASF1B       ASF1B</span>
<span class="co">#&gt;  4 1-cell stage cell… CellM… Human   Embryo 1-cell st…      32 BCL2L10     BCL2…</span>
<span class="co">#&gt;  5 1-cell stage cell… CellM… Human   Embryo 1-cell st…      32 BLCAP       BLCAP</span>
<span class="co">#&gt;  6 1-cell stage cell… CellM… Human   Embryo 1-cell st…      32 CASC3       CASC3</span>
<span class="co">#&gt;  7 1-cell stage cell… CellM… Human   Embryo 1-cell st…      32 CLEC10A     CLEC…</span>
<span class="co">#&gt;  8 1-cell stage cell… CellM… Human   Embryo 1-cell st…      32 CNOT11      CNOT…</span>
<span class="co">#&gt;  9 1-cell stage cell… CellM… Human   Embryo 1-cell st…      32 DCLK2       DCLK2</span>
<span class="co">#&gt; 10 1-cell stage cell… CellM… Human   Embryo 1-cell st…      32 DHCR7       DHCR7</span>
<span class="co">#&gt; # … with 422,282 more rows</span></code></pre></div>
<p>Check the number of available cell types.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">markers</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html">distinct</a></span><span class="op">(</span><span class="va">celltype_full</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; [1] 3039</span></code></pre></div>
<p>Check the number of available cell types per species (not available for every cell type).</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">markers</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html">distinct</a></span><span class="op">(</span><span class="va">celltype_full</span>, <span class="va">species</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/count.html">count</a></span><span class="op">(</span><span class="va">species</span>, sort <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 3 x 2</span>
<span class="co">#&gt;   species     n</span>
<span class="co">#&gt;   &lt;chr&gt;   &lt;int&gt;</span>
<span class="co">#&gt; 1 "Human"  1866</span>
<span class="co">#&gt; 2 "Mouse"   850</span>
<span class="co">#&gt; 3 ""        323</span></code></pre></div>
<p>Check the number of available cell types per organ (not available for every cell type).</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">markers</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html">distinct</a></span><span class="op">(</span><span class="va">celltype_full</span>, <span class="va">organ</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/count.html">count</a></span><span class="op">(</span><span class="va">organ</span>, sort <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 93 x 2</span>
<span class="co">#&gt;    organ                  n</span>
<span class="co">#&gt;    &lt;chr&gt;              &lt;int&gt;</span>
<span class="co">#&gt;  1 ""                  2160</span>
<span class="co">#&gt;  2 "Brain"              122</span>
<span class="co">#&gt;  3 "Immune system"       50</span>
<span class="co">#&gt;  4 "Lung"                47</span>
<span class="co">#&gt;  5 "Kidney"              43</span>
<span class="co">#&gt;  6 "Bone marrow"         42</span>
<span class="co">#&gt;  7 "Liver"               38</span>
<span class="co">#&gt;  8 "Blood"               33</span>
<span class="co">#&gt;  9 "Embryo"              30</span>
<span class="co">#&gt; 10 "Peripheral blood"    29</span>
<span class="co">#&gt; # … with 83 more rows</span></code></pre></div>
<p>Check package version.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/packageDescription.html">packageVersion</a></span><span class="op">(</span><span class="st">"clustermole"</span><span class="op">)</span>
<span class="co">#&gt; [1] '1.0.1.9001'</span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Igor Dolgalev.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
